### DISCUSSION TITLE
Can't sign up to passkeys.io

### DISCUSSION THREAD

**Discussion by sammcj** [teamhanko/hanko#342] [Q&A] [answered]
Trying out passkeys.io seems to require you to own two phones.

When you visit the site (on your phone) and try to sign up/in it shows a pop-up with a QR code that needs to be scanned - so you can't scan it with the device you're using.

Here is a video of the user experience: https://user-images.githubusercontent.com/862951/197876176-af066b20-a800-4a59-85ed-2f1b697e5a52.MOV

### Reproducing the bug

1. Visit passkeys.io on an iPhone running iOS 16.1.
2. Try to sign in.
3. Get offered a QR to scan.

### OS Version

iOS 16.1

**Comment by FlxMgdnz** [comment, marked as answer]
Even though the "Sign in with a passkey" button looks like a typical SSO/OAuth button (e.g., "Sign in with Google"), passkeys work completely different, much more like a password in that sense.

1. Before being able to sign in with a passkey, you first need to create (a) an account on the page to sign in to, and (b) a passkey for that account
2.  The passkey creation is part of the onboarding flow when creating a new account
3. Since a passkey also contains the username, the login screen can't know who you are and if you have an account until the passkey is presented
4. Now, in your case, you did not have a passkey for passkeys.io on your device, so the UI anticipates a passkey from another device, either through the QR code hybrid flow, or from a physical Security key – that's just how it's supposed to work

That the "Sign in with a passkey" button may be confusing to many due to its similarities with OAuth / social login buttons makes perfect sense, though.

That's why passkey autofill ("Conditional UI") was developed. Over here in the Hanko project, we support this already. We'll bring it to passkeys.io very soon, too. With passkey autofill, the passkey button is not required anymore, and instead your browser is able to check if you have a passkey on your device, and only then shows you the option to use the passkey when you tap into the username field.

![image](https://user-images.githubusercontent.com/20115649/198101230-95ade889-ecb2-4017-afd3-a0475318c7bd.png)

**Comment by sammcj** [comment]
Thank you for the detailed response!

It certainly is confusing at present (at least on passkeys.io). I agree that if I was offered only options I could actually use from the device I was on - that would make the experience far less confusing.

**Comment by nickfujita** [comment]
Ran into the same issue when selecting passkey from iOS device, but was able to fix this in my iOS settings. Settings > Passwords > Password Options > USE PASSWORDS AND PASSKEYS FROM > (select iCloud Keychain)

In my case I have other providers and I guess had unchecked iCloud Keychain at some point. So the passkey creation worked for my other provider but not for iOS since it wasn't enabled. After enabling it in the settings it worked fine.
